utils/hare-shutdown

```diff
@@ -17,217 +17,78 @@
-    else:
-        logging.error(f'**ERROR** Failed to stop {proc.systemd_name}{label}'
-                      f' at {proc.node}')
 
-
-class StopProcess:
-    def __init__(self, process: Process):
-        self.process = process
-
-
-class QuitMessage:
-    pass
-
-
-class Worker(threading.Thread):
-    def __init__(self, queue: Queue):
-        super().__init__(target=self._do_work, args=(queue, ))
-
-    def _do_work(self, queue):
-        logging.debug('Started thread')
-        while True:
-            msg = queue.get()
-            if isinstance(msg, StopProcess):
-                process_stop(msg.process)
-            if isinstance(msg, QuitMessage):
-                logging.debug('Exiting thread')
-                return
-
-
-def _stop_parallel(process_list: List[Process], thread_count: int = 8) -> None:
-    size = len(process_list)
-
-    if not size:
-        return
-    q: Queue = Queue(maxsize=size)
-    thread_count = min(thread_count, size)
-
-    worker_pool: List[Worker] = [Worker(q) for i in range(thread_count)]
-
-    for w in worker_pool:
-        w.start()
-
-    for proc in process_list:
-        q.put(StopProcess(process=proc))
-    for i in range(thread_count):
-        q.put(QuitMessage())
-
-    for worker in worker_pool:
-        worker.join()
-
-
-def main() -> None:
-    _setup_logging()
-    if not consul_is_active_at('localhost'):
-        if pcs_consul_is_active_at('localhost'):
+def check_cluster():
+    if not utils.consul_is_active_at('localhost'):
+        if utils.pcs_consul_is_active_at('localhost'):
             exit("Cluster is run by Pacemaker, try `pcs cluster stop --all'")
         else:
             exit('Cluster is not running')
 
+
+@repeat_if_fails()
+def _get_processes_and_node():
     cns = Consul()
+    processes = utils.processes_by_consul_svc_name(cns)
+    leader_node = utils.get_kv(cns, 'leader')
+    return (processes, leader_node)
 
-    @repeat_if_fails()
-    def _get_processes_and_node():
-        processes = processes_by_consul_svc_name(cns)
-        leader_node = get_kv(cns, 'leader')
-        return (processes, leader_node)
 
+def shutdown_cluster():
+    check_cluster()
     (processes, leader_node) = _get_processes_and_node()
-    for svc in shutdown_sequence:
+    for svc in utils.shutdown_sequence:
         # Some processes may be not started as per configuration
         if svc in processes:
-            _stop_parallel([proc for proc in processes[svc]])
+            utils.stop_parallel([proc for proc in processes[svc]])
 
-    if leader_node and not is_fake_leader_name(leader_node):
+    if leader_node and not utils.is_fake_leader_name(leader_node):
         logging.info(f'Shutting down RC Leader at {leader_node}... ')
         pkill: str = 'sudo pkill --exact -KILL consul &> /dev/null'
-        exec(ssh_prefix(leader_node) + pkill)
+        utils.exec_custom(utils.ssh_prefix(leader_node) + pkill)
+
+
+def local_node_shutdown():
```

Comment:
- rename `local_node_shutdown()` to `node_shutdown()`
- take `process` as an argument
-  invoke `node_shutdown()` from shutdown_cluster()
-  pass `process` to `node_shutdown()` from main if `--node` is passed.

Reply:
Changed the name for local_node_shutdown()
added stop_process() which will take process as arguments and stop the process.
 stop_process() is called from node_shutdown() and shutdown_cluster()